src/Sylius/Bundle/CoreBundle/Form/EventSubscriber/AddAuthorGuestTypeFormSubscriber.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Form\EventSubscriber;
+
+use Sylius\Component\Review\Model\ReviewInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class AddAuthorGuestTypeFormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'preSetData',
+        ];
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $form = $event->getForm();
+        /** @var ReviewInterface $review */
+        $review = $event->getData();
+        $author = $form->getConfig()->getOption('author');
+
+        if (null === $author && null === $review->getAuthor()) {
```

Comment:
There are two places first one is the review and second one is current logged in user. Maybe i should change ``$author`` variable name to something like ``$currentLoggedAuthor``. WDYT?